src/__tests__/client.spec.ts

```diff
@@ -51,4 +62,65 @@ describe("API Client", function (): void {
     client.setTimeouts(30000);
     expect(client.config.connectionTimeoutMillis).toBe(30000);
   });
+
+  test("should throw error if environment is not defined", () => {
+    expect(() => {
+      new Client({ apiKey: "ADYEN_API_KEY" } as any);
+    }).toThrow("Environment must be defined");
+  });
+
+  test("should throw error if environment is LIVE and region is invalid", () => {
+    const config = new Config({
+      apiKey: "ADYEN_API_KEY",
+      environment: "LIVE",
+      region: "INVALID" as RegionEnum,
+      liveEndpointUrlPrefix: "prefix"
+    });
+    expect(() => {
+      new Client(config);
+    }).toThrow("Invalid region provided: INVALID");
+  });
+
+  test("should set terminalApiCloudEndpoint for TEST region", () => {
+    const config = new Config({
+      apiKey: "ADYEN_API_KEY",
+      environment: "TEST"
+    });
+    const client = new Client(config);
+    expect(client.config.terminalApiCloudEndpoint).toBeDefined();
+    expect(client.config.terminalApiCloudEndpoint).toBe("https://terminal-api-test.adyen.com");
+  });
+
+  test("should set terminalApiCloudEndpoint for LIVE region", () => {
+    const config = new Config({
+      apiKey: "ADYEN_API_KEY",
+      environment: "LIVE",
+      region: RegionEnum.US,
+      liveEndpointUrlPrefix: "prefix"
```

Comment:
just to confirm: `region` will only affect Terminal API's url, `liveEndpointUrlPrefix` will only affect Checkout/PAL APIs. Technically can instantiate one client and pass it to both Terminal API and Checkout API constructors right?